src/json/from_json.cr

```diff
@@ -237,7 +237,7 @@ def NamedTuple.new(pull : JSON::PullParser)
 
     {
       {% for key, type in T %}
-        {{key}}: (%var{key.id}).as({{type}}),
+        "{{key}}": (%var{key.id}).as({{type}}),
```

Comment:
I'm almost sure that if the key has a quote, it will break. You can use dot id dot stringify instead. I would also add a spec for this

Reply:
I think I found a diff bug:

```cr
require "json"

JSON.parse %({"foo "5" bar": 1}) # => Unhandled exception: unexpected token '5' at line 1, column 8 (JSON::ParseException)
```

It seems that neither YAML or JSON parsers can handle quotes within a key.